accounts-db/src/read_only_accounts_cache.rs

```diff
@@ -64,33 +68,66 @@ impl ReadOnlyCacheStats {
 
 #[derive(Debug)]
 pub(crate) struct ReadOnlyAccountsCache {
-    cache: DashMap<ReadOnlyCacheKey, ReadOnlyAccountCacheEntry>,
+    cache: Arc<DashMap<ReadOnlyCacheKey, ReadOnlyAccountCacheEntry>>,
     /// When an item is first entered into the cache, it is added to the end of
     /// the queue. Also each time an entry is looked up from the cache it is
     /// moved to the end of the queue. As a result, items in the queue are
     /// always sorted in the order that they have last been accessed. When doing
     /// LRU eviction, cache entries are evicted from the front of the queue.
-    queue: Mutex<IndexList<ReadOnlyCacheKey>>,
-    max_data_size: usize,
-    data_size: AtomicUsize,
+    queue: Arc<Mutex<IndexList<ReadOnlyCacheKey>>>,
+    _max_data_size_lo: usize,
+    max_data_size_hi: usize,
+    data_size: Arc<AtomicUsize>,
     // read only cache does not update lru on read of an entry unless it has been at least this many ms since the last lru update
     ms_to_skip_lru_update: u32,
 
     // Performance statistics
-    stats: ReadOnlyCacheStats,
+    stats: Arc<ReadOnlyCacheStats>,
     highest_slot_stored: AtomicU64,
+
+    /// Channel to send eviction requests
+    ///
+    /// NOTE: This field must be above `evictor` to ensure it is dropped before `evictor`.
```

Comment:
The comment seems confusing. I thought the fields are destructed backward?
_evictor drop first then evict_sender?

Reply:
oh. interesting. That's different from C++, which destruct in reverse order.